### PR TITLE
Configure Grafana with OpenShift login

### PR DESCRIPTION
**Pull request by avi-biton**
- Configure Grafana with OpenShift login for all environments
- Add example dashboard example

### CODE REVIEW

**Comment by avi-biton** [comment]
@gbenhaim @amisstea Please review.

File: components/monitoring/grafana/base/grafana-app.yaml [hunk 1]
```diff
@@ -1,11 +1,25 @@
-# Create service account 
+# Create service account to add Prometheus DataDource
```
Comment (amisstea) [inline components/monitoring/grafana/base/grafana-app.yaml:1]:
```suggestion
# Create service account to add Prometheus Datasource 
```

File: components/monitoring/grafana/base/grafana-app.yaml [hunk 2]
```diff
@@ -1,11 +1,25 @@
-# Create service account 
+# Create service account to add Prometheus DataDource 
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: metrics-reader
   namespace: appstudio-grafana
 ---
-# Create secret for the service account
+# Create RBC for the metrics-reader service account
```
Comment (amisstea) [inline components/monitoring/grafana/base/grafana-app.yaml:8]:
```suggestion
# Create CRB for the metrics-reader service account
```

File: components/monitoring/grafana/base/grafana-app.yaml [hunk 3]
```diff
@@ -15,52 +29,175 @@ metadata:
...
-      mode: console
+      mode: "console"
+      level: "warn"
+    auth.anonymous:
+      enabled: True
+    auth:
+      disable_login_form: False
+      disable_signout_menu: True
+    auth.basic:
+      enabled: True
```
Comment (amisstea) [inline components/monitoring/grafana/base/grafana-app.yaml:113]: Is this enabled so the auth proxy can use it?

Comment (avi-biton) [reply components/monitoring/grafana/base/grafana-app.yaml:113]: Yes, we need it for internal use of Grafana.
It was suggested to us by the Grafana Operator contributors

File: components/monitoring/grafana/base/grafana-app.yaml (hunk 1 above)
Comment (avi-biton) [reply components/monitoring/grafana/base/grafana-app.yaml]: Done

File: components/monitoring/grafana/base/grafana-app.yaml (hunk 2 above)
Comment (avi-biton) [reply components/monitoring/grafana/base/grafana-app.yaml]: Done

**Review by amisstea — APPROVED** [review]

**Comment by yftacherzog** [comment]
lgtm

**Pull request merged**